template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/atlas/AtlasKnoxRoleConfigProvider.java

```diff
@@ -38,11 +76,58 @@ public List<String> getRoleTypes() {
         return List.of("ATLAS_SERVER");
     }
 
-    @Override
-    public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
+    private boolean isAtlasKnoxConfigurationNeeded(TemplatePreparationObject source) {
         return Objects.nonNull(source.getGatewayView())
                 && Objects.nonNull(source.getGatewayView().getExposedServices())
                 && source.getGatewayView().getExposedServices().contains(exposedServiceCollector.getAtlasService().getKnoxService());
     }
 
+    private boolean isSDXOptimizationEnabled(TemplatePreparationObject source) {
+        return source.getStackType().equals(StackType.DATALAKE)
+                && entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())
+                && isDatalakeVersionSupported(source.getBlueprintView().getBlueprintText());
+    }
+
+    private boolean isDatalakeVersionSupported(String blueprintText) {
+        if (StringUtils.isEmpty(blueprintText)) {
+            throw new IllegalArgumentException("Blueprint is empty.");
+        }
+        String versionString;
+        try {
+            versionString = objectMapper.readValue(blueprintText, CdhVersion.class).getCdhVersion();
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        LOG.info("incoming version: {}", versionString);
+        return isShapeVersionSupportedByRuntime(versionString);
+    }
+
+    private boolean isShapeVersionSupportedByRuntime(String runtime) {
+        Comparator<Versioned> versionComparator = new VersionComparator();
+        return versionComparator.compare(() -> runtime, () -> MEDIUM_DUTY_REQUIRED_VERSION) > -1;
+    }
```

Comment:
Is this functionality added to ensure the configuration is added only when the runtime is >=7.2.8?

I see two issues
1. Fix the variable name "MEDIUM_DUTY_REQUIRED_VERSION"
2. isDatalakeVersionSupported should not be triggered within isSDXOptimizationEnabled. 
What do you think>

Reply:
I do not know how the entitlement works, when, and how we enable it. 
But I do not check lower versions. 
So Before We remove the version check, I have to check some lower versions.

Reply:
Consider "MIN_VERSION_FOR_DIFFERENCIAL_BACKUP" instead of MEDIUM_DUTY_REQUIRED_VERSION
isDatalakeVersionSupported should be invoked inside isSDXOptimizationEnabled, instead, it should be called separately.